### DISCUSSION TITLE
grouping-mapping-widget mappingsClient createGroupProperty

### DISCUSSION THREAD

**Discussion by Paolo-lab** [iTwin/itwinjs-core#7265] [Q&A] [answered]
hello good evening everyone
my client has problems with odata using grouping-mapping-widget this is the thread I opened for problem,
#7243
To try to solve the problem since the installed version is 0.22 on my app was really old so I updated my package with version 0.33 .

Now I have some difficulties because in this version it is not possible to use
 **mappingsClient.createGroupProperty**
and **IMappingsClient.getGroupProperties**
respectively to create properties within a group and the second to extract them, know tell me how I can solve the problem

Thanks you are precious

**Comment by Guillar1** [comment, marked as answer]
Hey @Paolo-lab,

The `grouping-mapping-widget` re-exports the `@itwin/insights-client`, which is where those operations originate. It was updated in this PR: https://github.com/iTwin/viewer-components-react/pull/1041.

Over the past year, the clients were refactored to work with the new API version. You can see the details here: [Changelog](https://github.com/iTwin/insights-client/blob/main/CHANGELOG.md#minor-7). The methods you're referring to were moved into this new client: [PropertiesClient.ts](https://github.com/iTwin/insights-client/blob/main/src/grouping-and-mapping/clients/PropertiesClient.ts).

Note that in this version, the different types of properties have been merged and can now be defined under a single property. However, the functionality of each property type remains unchanged. For more information, you can refer to the documentation here: [Grouping and Mapping API - Create Property](https://developer.bentley.com/apis/grouping-and-mapping/operations/create-property/).
